dashboard/src/hooks/usePagination.tsx

```diff
@@ -13,6 +13,7 @@ export const usePagination = (
   const [endIndex, setEndIndex] = useState(0);
 
   useEffect(() => {
+    setStartIndex(0); //restart the index whenever there is a change in filters
```

Comment:
this won't work if the itemsPorPage and totalItems have the same count, needs to add `diffFilter` and `filter` to the dependency array

Reply:
you will need `useSearch` for this one

Reply:
Can you give me an example when it's possible? I tried with http://localhost:5173/tree/0bdf955f408e17b64d9dc4de09d3a2fc4f233aca?tableFilter=all&currentTreeDetailsTab=treeDetails.builds&diffFilter=%7B%7D changing the itemsPerPage and it doesn't seem to be broken

Reply:
it will be a rare case if not never, but it is possible since are primitives in javascript and the `useEffect` wont detect a change if they are the same but coming from another source